core/src/main/scala/io/snappydata/Literals.scala

```diff
@@ -144,35 +151,65 @@ object Property extends Enumeration {
       if (altName == null) name else name + '/' + altName
   }
 
-  type Type = ValueAlt
+  case class SQLValue(name: String, configEntry: SQLConfigEntry)
+      extends Property.Val(name) {
 
-  protected final def Val(name: String): ValueAlt =
-    new ValueAlt(name, null)
+    def defaultValue[T]: Option[T] = configEntry.defaultValue[T]
 
-  protected final def Val(name: String, prefix: String): ValueAlt =
-    new ValueAlt(name, prefix + name)
+    def defaultValueString: String = configEntry.defaultValueString
 
-  val Locators = Val(s"${Constant.STORE_PROPERTY_PREFIX}locators",
-    Constant.SPARK_PREFIX)
+    override def toString(): String = name
+  }
 
-  val McastPort = Val(s"${Constant.STORE_PROPERTY_PREFIX}mcast-port",
-    Constant.SPARK_PREFIX)
+  type Type = ValueAlt
 
-  val JobserverEnabled = Val(s"${Constant.JOBSERVER_PROPERTY_PREFIX}enabled")
+  type SQLType = SQLValue
```

Comment:
Please handle this case in LeadImpl snappyProperties variable. I am posting a modified code here because it won't let me comment on unmodified section of LeadImpl.

```scala
  private def extractPropertyKey(prop: String) = if (prop.startsWith(Constant
      .PROPERTY_PREFIX) && !prop.startsWith(Constant.STORE_PROPERTY_PREFIX)) {
    Some(prop.substring(Constant.PROPERTY_PREFIX.length))
  } else if (prop.startsWith(Constant.SPARK_SNAPPY_PREFIX) &&
      !prop.startsWith(Constant.SPARK_STORE_PREFIX)) {
    Some(prop.substring(Constant.SPARK_SNAPPY_PREFIX.length))
  } else {
    None
  }

  private[snappydata] val snappyProperties = Utils.getFields(Property).flatMap {
    case (_, propVal: Property.Type) =>
      extractPropertyKey(propVal.name)
    case (_, propVal: Property.SQLType) =>
      extractPropertyKey(propVal.name)
  }.toSet
```